lib/si-data-pg/src/lib.rs

```diff
@@ -2888,3 +2909,30 @@ impl PgOwnedTransaction {
 async fn test_connection_infallible_and_warm_up_pool_task(check_pool: PgPool) {
     let _result = check_pool.test_connection().await;
 }
+
+fn evict_old_and_idle_pgpool_connections(pg_pool: Pool, check_interval: Duration, max_age: Duration, max_idle_time: Duration) -> impl Future<Output = ()> {
+    let min_check_interval = Duration::from_secs(30);
+    let check_interval = check_interval.max(min_check_interval);
+
+    async move {
+        loop {
+            tokio::time::sleep(check_interval).await;
+            let connection_count = Rc::new(RefCell::new(0));
+            let evicted_connections = Rc::new(RefCell::new(0));
+            pg_pool.retain(|_, metrics| {
+                *connection_count.borrow_mut() += 1;
+                if metrics.last_used() > max_idle_time {
+                    debug!("Evicting connection due to idle time");
+                    *evicted_connections.borrow_mut() += 1;
+                    false
+                } else if metrics.age() > max_age {
+                    debug!("Evicting connection due to total age");
```

Comment:
should we `*evicted_connections.borrow_mut() += 1;` after this?

Reply:
fixed